hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/RequestConverter.java

```diff
@@ -1577,12 +1577,20 @@ public static IsMasterRunningRequest buildIsMasterRunningRequest() {
   }
 
   /**
-   * Creates a protocol buffer BalanceRequest
-   *
+   * Creates a protocol buffer BalanceRequest from a client BalanceRequest
+   * @param request
    * @return a BalanceRequest
    */
-  public static BalanceRequest buildBalanceRequest(boolean force) {
-    return BalanceRequest.newBuilder().setForce(force).build();
+  public static MasterProtos.BalanceRequest toBalanceRequest(BalanceRequest request) {
+    return MasterProtos.BalanceRequest.newBuilder()
+      .setDryRun(request.isDryRun())
+      .setIgnoreRit(request.isIgnoreRegionsInTransition())
+      .build();
+  }
+
+  public static BalanceRequest toBalanceRequest(MasterProtos.BalanceRequest request) {
+    BalanceRequest newRequest = request.getDryRun() ? BalanceRequest.dryRun() : BalanceRequest.execute();
```

Comment:
I think you need to use the `hasDryRun` and `hasIgnoreRit` methods to check if these fields are set on the `MasterProtos.BalanceRequest` instance, and then `getDryRun`, `getIgnoreRit` to access the values when they're present. Unless you explicitly want to rely on the protobuf default value behavior here... I'm not sure what that is, but I am pretty sure that's not a good idea from the perspective of long-term maintenance.

Reply:
The default for booleans is always false. I don't think this is necessary and feels somewhat redundant, but I'll add it in the next commit.